lib/streamlit/elements/button.py

```diff
@@ -292,7 +301,7 @@ def deserialize_button(ui_value: Optional[bool], widget_id: str = "") -> bool:
         download_button_proto.disabled = disabled
 
         self.dg._enqueue("download_button", download_button_proto)
-        return button_state.value
+        return cast(bool, button_state.value)
```

Comment:
This shouldn't be necessary, right? `register_widget` has been type-ified and should be returning a properly typed value. (Is this maybe just a result of the missing type annotation on `ButtonSerde.serialize`?